nutils/numeric.py

```diff
@@ -591,4 +591,51 @@ def levicivita(n: int, dtype=float):
     result[tuple(I)] = 1 - 2*(nperms % 2)
     return result
 
+
+def sinc(x, n=0):
+    '''
+    Evaluates the n-th derivative of the unnormalized sinc function:
+
+        sinc(x) = sin(x) / x
+    '''
+
+    x = numpy.asarray(x)
+    f = numpy.sinc(x / numpy.pi) # Numpy's implementation is normalized to π
+    if n == 0:
+        return f
+    # Derivatives are evaluated using either a recurrence relation or a Taylor
+    # series expansion, depending on proximity to the origin.
+    m = x >= 1
+    if m.any(): # points outside unit ball
+        # sinc'i(x) = (sin'i(x) - i sinc'i-1(x)) / x
+        fm = f[m]
+        xm = x[m]
+        for i in range(1, n+1): # traverse derivatives
+            fm *= -i
+            fm += [numpy.sin, numpy.cos][i%2](xm) * [1, 1, -1, -1][i % 4]
+            fm /= xm
+        if not f.shape:
+            return float(fm)
```

Comment:
The cast to `float` is wrong if `x` is complex. Besides the early return, is there a reason for this branch?

Reply:
fixed